testsuite/features/step_definitions/command_steps.rb

```diff
@@ -60,6 +60,31 @@
   node.run("rpm -q #{package}; test $? -ne 0")
 end
 
+Then(/^I wait for "([^"]*)" to be (uninstalled|installed) on "([^"]*)"$/) do |package, status, host|
+  if package.include?("suma") && $product == "Uyuni"
+    package.gsub! "suma", "uyuni"
+  end
+  node = get_target(host)
+  if host.include? 'ubuntu'
+    node.wait_while_process_running('apt-get')
+    pkg_version = package.split('-')[-1]
+    pkg_name = package.delete_suffix("-#{pkg_version}")
+    pkg_version_regexp = pkg_version.gsub('.', '\\.')
+    if status == 'installed'
+      node.run_until_ok("dpkg -l | grep -E '^ii +#{pkg_name} +#{pkg_version_regexp} +'")
+    else
+      node.run_until_fail("dpkg -l | grep -E '^ii +#{pkg_name} +#{pkg_version_regexp} +'")
+    end
+  else
+    node.wait_while_process_running('zypper')
```

Comment:
OK we go that way. Fine.

Reply:
Yep, but just because that step is strictly related to the action of installation so the command tool to install packages. 
So here, we want to make sure that the installation process is fully completed.
Otherwise, yes, as we discussed... if we do that somewhere else it will hide the issue and is not what we want.

Reply:
Exactly.